source/iterators.tex

```diff
@@ -719,31 +719,34 @@
 The type \tcode{iter_difference_t<I>} denotes
 \begin{itemize}
 \item
-\tcode{incrementable_traits<I>::difference_type}
-if \tcode{iterator_traits<I>} names a specialization
+\tcode{incrementable_traits<J>::difference_type}
+if \tcode{iterator_traits<J>} names a specialization
 generated from the primary template, and
 
 \item
-\tcode{iterator_traits<I>::\brk{}difference_type} otherwise.
+\tcode{iterator_traits<J>::difference_type} otherwise,
 \end{itemize}
+where \tcode{J} is \tcode{remove_cvref_t<I>}.
```

Comment:
It's a bit unfortunate that we introduce J so late, after all the bullets. Can we introduce it before the bulleted list?

Reply:
Do you have a concrete suggestion? I don't want to break up the flow of this paragraph or make it too different from the phrasing used in parallel sentences elsewhere.

One possibility:

> The type `iter_difference_t<I>` denotes
> * `iter_difference_t<remove_cvref<I>>` if `remove_cvref<I>` is not the same type as `I`, or
> * `incrementable_traits<I>::difference_type` if `iterator_traits<I>` names a specialization generated from the primary template, or
> * `iterator_traits<I>::difference_type` otherwise.

What do you think?

Reply:
We don't seem to constraint the template parameter "I" at all, and we seem to use "class T" in the synopsis instead of I.
If these assertions hold, maybe something like "For a type I, where J denotes remove_cvref&lt;I>, the type iter_difference_t&lt;I> denotes... (text as shown in ingress paper)."